scripts/pre_commit_linter.py

```diff
@@ -1322,17 +1322,27 @@ def _check_html_directive_name(all_files):
     return summary_messages
 
 
-def _validate_and_parse_js_file(filename, content):
+def _validate_and_parse_js_file(all_files):
     """This function validates a JavaScript file and returns the parsed contents
     as a Python dictionary.
     """
+    parsed_js_files_dict = dict()
     # Use Pyjsparser to parse a JS file as a Python dictionary.
     parser = pyjsparser.PyJsParser()
-    print 'Validating and parsing %s file ...' % filename
-    return parser.parse(content)
+    # Select JS files which need to be checked.
+    files_to_check = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)
+        and filename.endswith('.js')]
+    for filename in files_to_check:
```

Comment:
Use the existing file caching here (see [this line](https://github.com/oppia/oppia/pull/6125/files#diff-3093ff31fd2fc587c0d99fca7d3afdf0L1350))

Reply:
Done